pkg/cache/partition_info.go

```diff
@@ -715,55 +715,46 @@ func checkAndSetResource(resource *resources.Resource) string {
 	if resource != nil {
 		return strings.Trim(resource.String(), "map")
 	}
-	return ""
+	return "[]"
 }
 
-// TODO fix this:
-// should only return one element, only a root queue
-// remove hard coded values and unknown AbsUsedCapacity
-// map status to the draining flag
-func (pi *PartitionInfo) GetQueueInfos() []dao.QueueDAOInfo {
-	pi.RLock()
-	defer pi.RUnlock()
-
-	var queueInfos []dao.QueueDAOInfo
-
-	info := dao.QueueDAOInfo{}
-	info.QueueName = pi.Root.Name
-	info.Status = pi.Root.stateMachine.Current()
-	info.Capacities = dao.QueueCapacity{
-		Capacity:        checkAndSetResource(pi.Root.GetGuaranteedResource()),
-		MaxCapacity:     checkAndSetResource(pi.Root.GetMaxResource()),
-		UsedCapacity:    checkAndSetResource(pi.Root.GetAllocatedResource()),
-		AbsUsedCapacity: "20",
-	}
-	info.ChildQueues = GetChildQueueInfos(pi.Root)
-	queueInfos = append(queueInfos, info)
-
-	return queueInfos
-}
-
-// TODO fix this:
-// should only return one element, only a root queue
-// remove hard coded values and unknown AbsUsedCapacity
-// map status to the draining flag
-func GetChildQueueInfos(info *QueueInfo) []dao.QueueDAOInfo {
-	var infos []dao.QueueDAOInfo
-	for _, child := range info.children {
-		queue := dao.QueueDAOInfo{}
-		queue.QueueName = child.Name
-		queue.Status = child.stateMachine.Current()
-		queue.Capacities = dao.QueueCapacity{
-			Capacity:        checkAndSetResource(child.GetGuaranteedResource()),
-			MaxCapacity:     checkAndSetResource(child.GetMaxResource()),
-			UsedCapacity:    checkAndSetResource(child.GetAllocatedResource()),
-			AbsUsedCapacity: "20",
+func calculateAbsUsedCapacity(capacity *resources.Resource, usedCapacity *resources.Resource) *resources.Resource {
+	if capacity == nil {
+		return nil
+	}
+	if usedCapacity == nil {
+		return resources.NewResource()
+	}
+	absResource := make(map[string]resources.Quantity)
+	for resourceName, availableResource := range capacity.Resources {
+		var absResValue float32
+		if availableResource != 0 {
+			if usedResource, ok := usedCapacity.Resources[resourceName]; ok {
+				absResValue = float32(usedResource) / float32(availableResource) * 100
+			}
 		}
-		queue.ChildQueues = GetChildQueueInfos(child)
-		infos = append(infos, queue)
+		absResource[resourceName] = resources.Quantity(absResValue)
 	}
+	return resources.NewResourceFromMap(absResource)
+}
 
-	return infos
+func (qi *QueueInfo) GetQueueInfos() dao.QueueDAOInfo {
+	qi.RLock()
+	defer qi.RUnlock()
```

Comment:
Now that this is part of the QueueInfo it should be moved to the QueueInfo file.
I know it is allowed to do this but maintenance will become really difficult when we leave it here.

NOTE: tests should also be moved to the correct file and if no tests are available we need a test for this.

Reply:
Moved it + also added a small unit test for it